Feat / Add periodic task refetching to campaign page

royschut opened the pull request:
This PR:
- adds periodic refetching once there are no tasks (anymore)
- adds filtering to make sure submitted tasks aren't refetched
- refactoring: uses useLazyQuery for fetching instead of client.query for better readability

royschut committed:
feat(campaign): add periodic task refetching

RCVZ reviewed (approved):
Nice!

VincentCouzij commented on src/containers/Task/Task.jsx:
```diff
@@ -2,139 +2,102 @@ import React, { useState, useEffect } from 'react';
+  const [tasksTodo, setTasksTodo]           = useState([]);
+  const [tasksSubmitted, setTasksSubmitted] = useState([]);
+  const taskDrawAmount                      = 10;
+  const refetchTimeout                      = 3000;
```
Is this not a bit much? do we expect new tasks to be added every 3 seconds?

royschut replied:
That was for testing purposes, but thanks for reminding me. 30000 it is!

royschut committed:
chore(campaign): add more readability

- combine useEffects
- set refetchTimeout to 30s
- set query params in function call
- fix not working loading state
- use the drawAmount in the query

demmyhonore commented on src/containers/Task/Task.jsx:
```diff
@@ -2,139 +2,95 @@ import React, { useState, useEffect } from 'react';
+  const [tasksSubmitted, setTasksSubmitted] = useState([]);
+  const taskDrawAmount                      = 10;
+  const refetchTimeout                      = 30000;
+  const [refetchCount, setRefetchCount]     = useState(0);
```
Are refetchCount and taskDrawAmount doing two different things?

royschut replied:
Yes. The taskDrawAmount is the maximum amount of tasks fetched. The refetchCount is a counter that adds up on every refetch timeout (30secs), so the useEffect with the fetch gets retriggered.

demmyhonore commented on src/containers/Task/Task.jsx:
```diff
@@ -2,139 +2,95 @@ import React, { useState, useEffect } from 'react';
-    if (taskIdentifier || loading || !campaign) {
-      return;
+    if (!taskIdentifier && score) {
+      getTaskList({ variables: { identifier: score.identifier, filtered: tasksSubmitted, drawamount: taskDrawAmount } });
```
Is it required to make drawamount camelCasing here?

royschut replied:
Good point. It's not required, but it's allowed, so it makes sense to keep consistency

demmyhonore reviewed (approved):
Nice refactor, a few minor questions.

royschut committed:
chore(campaign): fix camelcase
